### ISSUE TITLE
Did you notice that the loss in the training process is negative ?

### ISSUE THREAD

**Issue by CoinCheung** [ZhenyueQin/Research-Softmax-with-Mutual-Information#2]
Hi, 

I am so interested in the research you did, and I tried to run the code you released. I just found that with default configuration, the loss printed on the screen are negative numbers. when I changed this line: 
https://github.com/ZhenyueQin/Research-Softmax-with-Mutual-Information/blob/5c77c0bda6d5652e2d525e1acf1606d67d047cfd/pc_softmax_higher_accuracy/mnist_experiments/classification_mnist.py#L67

with this:  
```python
return torch.log_softmax(x, dim=1)
```
The loss values start to become positive. Have you observed this phenomenon ?

**Comment by kfzyqin** [comment]
Hi, 

Thanks for your interests! 

We are aware that the loss is negative. The negativity is because the loss can be considered as the negative pointwise mutual information between data and label. Such a value is intrinsically negative. We know this is anti-intuitive. 

If you don't mind a little bit more tedious comments, traditional softmax can be considered as the conditional entropy H(y | x), where x is the input data and y is the label. If we don't consider the continuous case, entropy is always positive, and the traditional softmax can be considered as minimising H(y | x).

**Comment by CoinCheung** [comment]
Thanks for replying !! Will this softmax score range (can be greater than 1) affect the function of cross entropy ? Or does cross entropy requires the input scores to be within [0, 1] to provide effective gradients for training ?

**Comment by kfzyqin** [comment]
We did not use the cross-entropy loss. Instead, we use negative log-likelihood (NLL). This is different from cross-entropy because cross-entropy involves a traditional softmax. That is, cross-entropy(x) = NLL(softmax(x)). In our case, we replace the (traditional) softmax with pc-softmax. 

People generally use softmax to make values in the range of (0, 1) in order to consider them as probabilities. In our case, the output value is negative pointwise mutual information, so the range changed.
